### DISCUSSION TITLE
Import stories from an external package

### DISCUSSION THREAD

**Discussion by andreilupu-plt** [storybookjs/react-native#443] [Q&A]
Are there any solution to import the stories from a package?

I'd like to have a package (e.g. **package-a**) and into this package to have some reusable components & stories from those components. Into my app, I'd like to import those stories too.

Is that possible?

**Comment by dannyhw** [comment]
Yes that should be possible, you would need to add the package folder to your metro watch folders though so that those files get resolved by metro.

If you share more details about your setup I can try to advise further

**Comment by andreilupu-plt** [comment]
Thank you for your quick reply. I could create a project a demo project if you need that.

Long story short: I have a package (e.g. `mobile-design-system`) Where I have a lot of components which a lot of reusable components:

`src > components`
 
```
AppBanner.tsx
AppPicket.tsx
Button.tsx
```

In `components` directory I have `stories` 
```
AppBanner.stories.tsx
AppPicket.stories.tsx
Button.stories.tsx
```

Currently:
- the stories are generate from this package and uploaded into chromatic and are available into web
- the package exports only the components, not the stories, but I want it to do that because I have to load the storybook in the simulator from a different app


The package (`mobile-design-system`) has as dependencies:
```
   "@storybook/addon-actions": "^6.5.13",
    "@storybook/addon-essentials": "^6.5.13",
    "@storybook/addon-interactions": "^6.5.13",
    "@storybook/addon-links": "^6.5.13",
    "@storybook/builder-webpack4": "^6.5.13",
    "@storybook/manager-webpack4": "^6.5.13",
    "@storybook/react": "^6.5.13",
    "@storybook/testing-library": "0.0.13",
     "chromatic": "^6.17.0",
 ```

The app will have as dependencies the latest `@storybook/react-native` version available

**Reply by dannyhw** [reply]
Are you using a mono repo for this or are you getting the package from npm for example?

**Reply by andreilupu-plt** [reply]
I'm getting the package using `npm` 😊

P.S. Also, I'd like to have into that package an example directory with a demo app which loads those stories from that package (`mobile-design-system > example > storybookApp`)

**Reply by dannyhw** [reply]
Well as long as the package ships the stories alongside the components then it should be possible to target them.
